src/Container/Container.php

```diff
@@ -165,4 +222,44 @@ public function __call($name, $arguments)
             throw new \BadMethodCallException("No application method named {$name}()");
         }
     }
+
+    /**
+     * Instantiate an object of named class using reflection to recursively to resolve dependencies
+     *
+     * @param string $className
+     * @return mixed
+     * @throws \ReflectionException
+     */
+    protected function resolveInstance(string $className) {
```

Comment:
if this is the FQCN of the class maybe the doc comment of the parameter name should reflect this?